pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/client/Auth0ManagementClient.java

```diff
@@ -527,24 +529,43 @@ private ApiResult<String, APIException> tryPasswordChangeRequest(PasswordChangeT
         });
     }
 
-    private <B, E> ApiResult<B, E> withRetries(String retryMessage, Supplier<ApiResult<B, E>> callback) {
+    public <B, E> ApiResult<B, E> withRetries(String retryMessage, Supplier<ApiResult<B, E>> callback) {
         ApiResult<B, E> res = null;
         int numTries = 0;
         int maxTries = maxRetries + 1;
+        long totalSleepTimeMilliSecond = 0L;
         while (numTries < maxTries) {
             res = callback.get();
             numTries++;
             if (numTries >= maxTries) {
                 break;
             }
             if (res.getStatusCode() == 429) {
-                log.error(retryMessage, res.getError());
+                log.error(retryMessage, res.getError() + " " + res.getError().getClass().getName());
```

Comment:
Does this work? Are you getting any warnings?
It seems like you are concatenating an Throwable with a string, which may work via coercion, but does log.error accept that and provide readable output?
As I mentioned before, sticking this all in a method and writing a unit test for it will demonstrate whether it works.